lib/samlp.js

```diff
@@ -225,6 +225,7 @@ module.exports.auth = function(options) {
           RelayState:      opts.RelayState || (req.query || {}).RelayState || (req.body || {}).RelayState || '',
           SAMLResponse:    response.toString('base64')
         }));
+        next();
```

Comment:
if you call `res.send()`, it calls `res.end()` for you, so you could override it:

```
var end = res.end;
res.end = function(chunk, encoding) {
  // your asserts

  res.end = end;
  res.end(chunk, encoding);
};
```